### PR TITLE
Review backend

### PR DESCRIPTION
**Pull request by yminsky**

### CODE REVIEW

File: book/compiler-backend/README.md [hunk 1]
```diff
@@ -186,75 +185,86 @@ benchmarking]{.idx}
...
-  Name                         Time/Run   Percentage
- ---------------------------- ---------- ------------
-  Polymorphic pattern           15.72ns      100.00%
-  Monomorphic larger pattern     9.41ns       59.89%
-  Monomorphic small pattern      8.73ns       55.56%
-
+File "bench_patterns.ml", line 1:
+Error: The files /home/yminsky/Code/rwo/_build/install/default/lib/core/core.cmi
+       and /home/yminsky/Code/rwo/_build/install/default/lib/core_bench/core_bench__Bench.cmi
+       make inconsistent assumptions over interface Sexplib__Pre_sexp
```
Comment (avsm) [inline book/compiler-backend/README.md:266]: Something's gone wrong here

File: book/compiler-backend/README.md [hunk 2]
```diff
@@ -713,24 +723,13 @@ let () =
...
-
-  Name                       Time/Run   Percentage
- ------------------------ ------------ ------------
-  Polymorphic comparison   6_856.10ns      100.00%
-  Monomorphic comparison     753.02ns       10.98%
-
+File "bench_poly_and_mono.ml", line 1:
+Error: The files /home/yminsky/Code/rwo/_build/install/default/lib/core/core.cmi
+       and /home/yminsky/Code/rwo/_build/install/default/lib/core_bench/core_bench__Bench.cmi
+       make inconsistent assumptions over interface Sexplib__Pre_sexp
```
Comment (avsm) [inline book/compiler-backend/README.md:731]: and here

File: book/compiler-backend/README.md (hunk 1 above)
Comment (yminsky) [reply book/compiler-backend/README.md:266]: Indeed. That's why this is a draft. runtest produced those promotions. I was hoping you might have some idea why!

Comment (avsm) [reply book/compiler-backend/README.md]: fixed in 38836a9e876ea5643e183d82e688a6b57c78fcb0 -- was missing .rwo-example files for those two so it was picking up your global ppx_sexp

**Review by avsm — APPROVED** [review]
will merge once CI passes

**Pull request merged**